packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -0,0 +1,137 @@
+import { Provider } from "@ethersproject/abstract-provider";
+
+export enum OptimisticOracleType {
+  PriceRequest = "PriceRequest",
+  Assertion = "Assertion",
+}
+
+export class OptimisticOracleRequest {
+  body: string; // Human readable request body.
+  type: OptimisticOracleType; // Type of the request.
+  timestamp: number; // Timestamp in seconds of the request.
+  identifier: string; // Identifier of the request.
+  requester: string; // Address of the requester.
+  requestTx: string; // Transaction hash of the request.
+  proposer?: string; // Address of the proposer.
+  proposedValue?: number | boolean; // Proposed value.
+  proposeTx?: string; // Transaction hash of the proposal.
+  disputableUntil?: number; // Timestamp in ms until the request can be disputed.
+  resolvedValue?: number | boolean; // Resolved value.
+  resolveTx?: string; // Transaction hash of the resolution.
+  disputeTx?: string; // Transaction hash of the dispute.
+
+  constructor(data: {
+    body: string;
+    type: OptimisticOracleType;
+    timestamp: number;
+    identifier: string;
+    requester: string;
+    requestTx: string;
+    proposer?: string;
+    proposedValue?: number | boolean;
+    proposeTx?: string;
+    disputableUntil?: number;
+    resolvedValue?: number | boolean;
+    resolveTx?: string;
+    disputeTx?: string;
+  }) {
+    this.body = data.body;
+    this.type = data.type;
+    this.timestamp = data.timestamp;
+    this.identifier = data.identifier;
+    this.requester = data.requester;
+    this.requestTx = data.requestTx;
+    this.proposer = data.proposer;
+    this.proposedValue = data.proposedValue;
+    this.proposeTx = data.proposeTx;
+    this.disputableUntil = data.disputableUntil; // should be in ms
+    this.resolvedValue = data.resolvedValue;
+    this.resolveTx = data.resolveTx;
+    this.disputeTx = data.disputeTx;
+  }
+}
+
+/**
+ * Represents a client to interact with an Optimistic Oracle and store the requests.
+ */
+export class OptimisticOracleClient {
+  protected provider: Provider;
+  protected requests: OptimisticOracleRequest[];
+  protected fetchedBlockRange: [number, number];
+
+  /**
+   * Constructs a new OptimisticOracleClient instance.
+   * @param _provider The provider used for interacting with the blockchain.
+   * @param _requests (Optional) The list of Optimistic Oracle requests.
+   */
+  constructor(_provider: Provider, _requests: OptimisticOracleRequest[] = []) {
+    this.provider = _provider;
+    this.requests = _requests;
+    this.fetchedBlockRange = [0, 0];
```

Comment:
Should this be provided to the constructor as well?

Reply:
+1